### PR TITLE
Fix the event delta time and event number

### PR DESCRIPTION
**Pull request by yunhanw-google**
#### Problem
Fix delta time which needs to retrieved from event buffer when calling FetchEventSince.
Fix event number calculation

#### Change overview
See above

#### Testing
Manual run and validate the time and number.

### CODE REVIEW

File: src/app/EventManagement.cpp [hunk 1]
```diff
@@ -460,8 +460,8 @@ CHIP_ERROR EventManagement::CopyAndAdjustDeltaTime(const TLVReader & aReader, si
 
     if (aReader.GetTag() == TLV::ContextTag(to_underlying(EventDataIB::Tag::kPath)))
     {
-        err =
-            ctx->mpWriter->Put(TLV::ContextTag(to_underlying(EventDataIB::Tag::kEventNumber)), ctx->mpContext->mCurrentEventNumber);
+        err = ctx->mpWriter->Put(TLV::ContextTag(to_underlying(EventDataIB::Tag::kEventNumber)),
+                                 ctx->mpContext->mCurrentEventNumber - 1);
```
Comment (bzbarsky-apple) [inline src/app/EventManagement.cpp:464]: Please document why this random `- 1` is here.  Why is `ctx->mpContext->mCurrentEventNumber` not the right number?

And separately, _can_ it be made the right number?

Comment (yunhanw-google) [reply src/app/EventManagement.cpp:464]: yes, mCurrentEventNumber is increased to next Event Number when it found it is same priority with the one set by FetchEventSince.  After that, we start to encoding, where we need -1,   later mCurrentEventNumber is returned to report engine which is used as next starting event number.

Comment (yunhanw-google) [reply src/app/EventManagement.cpp:464]: Just introduce bool variable, then after finish encoding, we check if this boolean is true, then increase the event number for next use.

Comment (yunhanw-google) [reply src/app/EventManagement.cpp]: Done, thanks

File: src/app/EventManagement.cpp [hunk 2]
```diff
@@ -659,6 +659,10 @@ CHIP_ERROR EventManagement::EventIterator(const TLVReader & aReader, size_t aDep
     if (event.mPriority == apEventLoadOutContext->mPriority)
     {
         apEventLoadOutContext->mCurrentTime.mValue += event.mDeltaTime.mValue;
+        // If the retrieved event has same priority with the one set by FetchEventSince, we need use mSamePriorityEvent to increase
```
Comment (bzbarsky-apple) [inline src/app/EventManagement.cpp:662]:
```suggestion
        // If the retrieved event has the same priority as the one set by FetchEventSince, we need to use mSamePriorityEvent to increase
```

File: src/app/EventManagement.cpp [hunk 3]
```diff
@@ -659,6 +659,10 @@ CHIP_ERROR EventManagement::EventIterator(const TLVReader & aReader, size_t aDep
...
+        // event number after encoding current even number, then this updated event number would be used by FetchEventSince for next
```
Comment (bzbarsky-apple) [inline src/app/EventManagement.cpp:663]:
```suggestion
        // the event number after encoding the current event number.  Then this updated event number will be used by FetchEventSince for next
```

**Review by bzbarsky-apple — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

**Pull request merged**